akvo/codelists/models/base_codelist.py

```diff
@@ -10,8 +10,9 @@
 
 
 class BaseCodelist(models.Model):
-    code = models.CharField(_(u'code'), max_length=100, blank=True, null=False)
-    version = models.ForeignKey('Version', verbose_name=_(u'version'), blank=False, null=False)
+    code = models.CharField(_(u'code'), max_length=100, blank=True, null=False, db_index=True)
+    version = models.ForeignKey('Version', verbose_name=_(u'version'), blank=False, null=False,
+                                db_index=True)
```

Comment:
It looks like the **name** field could also be included in the abstract model, and possibly **description** too. Also the **__unicode__()** method might be declared here rather than in all subclasses as it looks like the implementation is the same for many if not all of them.

Maybe it even makes sense to create a few abstract base classes and use them as mixins.